src/test/java/com/napier/sem/DAO_IntegrationTests.java

```diff
@@ -231,4 +176,107 @@ void topNCountriesIn_allAreasMatchFilter()
         // then
         for (Country country: countries) assertEquals("Europe", country.getContinent());
     }
-}
\ No newline at end of file
+
+    /**
+     * Integration tests covering the DAO.allCitiesIn method
+     */
+    // Tests that all countries in query are in the given area
+    @Test
+    void allCitiesIn_allAreasMatchFilter()
+    {
+        // given
+        String areaFilter = "Caribbean";
+
+        // when
+        ArrayList<City> cities = dao.allCitiesIn(App.REGION, areaFilter);
+
+        // then
+        for (City city: cities) assertEquals(areaFilter, city.getCountry());
+    }
+
+    // Tests that an empty array is returned when the areaName is set incorrectly
+    @Test
+    void allCitiesIn_arrayIsEmptyWhenAreaNameIsInvalid() {
+        // given
+        String areaName = "Scotland";
```

Comment:
scotland is not an invalid name for a country

Reply:
i'm realising now that it might be, but it's not obvious so we should make this something that's more obviously wrong

Reply:
Yeah they have Scotland as a district I think. Can see the thinking though so happy to change it.

Reply:
should be fixed